docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,329 @@
+useful, we may promote it to a named field of ImageReviewSpec.
+
+In the case of a Pod update, Kubernetes may send the backend either all images in the updated image, or only the ones that
+changed, at its discretion.
+
+## Interaction with Controllers
+
+In the case of a Deployment object, no image check is done when the Deployment object is created or updated.
+Likewise, no check happens when the Deployment controller creates a ReplicaSet.  The check only happens
+when the ReplicaSet controller creates a Pod.  Checking Pod is necessary since users can directly create pods,
+and since third-parties can write their own controllers, which kubernetes might not be aware of or even contain
+pod templates.
+
+The ReplicaSet, or other controller, is responsible for recognizing when a 403 has happened
+(whether due to user not having permission due to bad image, or some other permission reason)
+and throttling itself and surfacing the error in a way that CLIs and UIs can show to the user.
+
+Issue [22298](https://github.com/kubernetes/kubernetes/issues/22298) needs to be resolved to
+propagate Pod creation errors up through a stack of controllers.
+
+## Changes in policy over time
+
+The Backend might change the policy over time.  For example, yesterday `redis:v1` was allowed, but today `redis:v1` is not allowed
+due to a CVE that just came out (fictional scenario).  In this scenario:
+. 
+
+- a newly created replicaSet will be unable to create Pods.
+- updating a deployment will be safe in the sense that it will detect that the new ReplicaSet is not scaling
+  up and not scale down the old one.
+- an existing replicaSet will not be unable to create Pods that replace ones which are terminated.  If this is due to
+  slow loss of nodes, then there should be time to react before significant loss of capacity.
+- For non-replicated things (size 1 ReplicaSet, PetSet), a single node failure may disable it.
+- a node rolling update will eventually check for liveness of replacements, and would be throttled if
+  in the case when the image was no longer allowed and so replacements could not be started.
+- rapid node restarts will cause existing pod objects to be restarted by kubelet.
+- slow node restarts or network partitions will cause node controller to delete pods and there will be no replacement
+
+It is up to the Backend implementor, and the cluster administrator who decides to use that backend, to decide
+whether the Backend should be allowed to change its mind.  There is a tradeoff between responsiveness
+to changes in policy, versus keeping existing services running.  The two models that make sense are:
+
+- never change a policy, unless some external process has ensured no active objects depend on the to-be-forbidden
+  images.
+- change a policy and assume that transition to new image happens faster than the existing pods decay.
```

Comment:
With changing policy there's also the question of what to do with currently running images? Should the cluster administrator be equipped with a tool to verify which are "in danger".